crates/admin/src/cluster_controller/service/state.rs

```diff
@@ -277,80 +281,779 @@ where
         Ok(())
     }
 
+    #[instrument(
+        level = "debug",
+        skip(self),
+        fields(
+            partition_table_version = ?self.partition_table_watcher,
+            logs_metadata_version = ?self.cluster_state_watcher.current().logs_metadata_version
+        ),
+    )]
     async fn trim_logs(&self) {
-        let result = self.trim_logs_inner().await;
+        let cluster_state = self.cluster_state_watcher.current();
+        let current_trim_points = match get_current_trim_points(&cluster_state, &self.bifrost).await
+        {
+            Ok(trim_points) => trim_points,
+            Err(err) => {
+                warn!("Failed to get current logs trim points. This can lead to increased log server disk usage: {err}");
+                return;
+            }
+        };
+
+        let new_trim_points = TrimMode::from(self.snapshots_repository_configured, cluster_state)
+            .safe_trim_points(&current_trim_points, Some(self.log_trim_threshold));
+        debug!(?new_trim_points, "Safe trim points");
 
-        if let Err(err) = result {
-            warn!("Could not trim the logs. This can lead to increased disk usage: {err}");
+        for (log_id, (trim_point, partition_id)) in new_trim_points {
+            info!(
```

Comment:
I'd try and avoid a per-log log message on levels like info, trace is good for those. For info, I'd expect a single log line (if you need to).